'test_bucket_w_default_event_based_hold' systest is flaky

From [this Kokoro failure](https://source.cloud.google.com/results/invocations/84ba5579-be63-4afe-994d-74c06394f44e/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-storage%2Fpresubmit%2Fpresubmit/log):

```python
__________ TestRetentionPolicy.test_bucket_w_default_event_based_hold __________

self = <test_system.TestRetentionPolicy testMethod=test_bucket_w_default_event_based_hold>

    def test_bucket_w_default_event_based_hold(self):
        from google.api_core import exceptions

        new_bucket_name = "w-def-ebh" + unique_resource_id("-")
        self.assertRaises(
            exceptions.NotFound, Config.CLIENT.get_bucket, new_bucket_name
        )
        bucket = retry_429_503(Config.CLIENT.create_bucket)(new_bucket_name)
        self.case_buckets_to_delete.append(new_bucket_name)

        bucket.default_event_based_hold = True
        bucket.patch()

        self.assertTrue(bucket.default_event_based_hold)
        self.assertIsNone(bucket.retention_period)
        self.assertIsNone(bucket.retention_policy_effective_time)
        self.assertFalse(bucket.retention_policy_locked)

        blob_name = "test-blob"
        payload = b"DEADBEEF"
        blob = bucket.blob(blob_name)
        blob.upload_from_string(payload)

        self.case_blobs_to_delete.append(blob)

        other = bucket.get_blob(blob_name)

        self.assertTrue(other.event_based_hold)
        self.assertFalse(other.temporary_hold)
        self.assertIsNone(other.retention_expiration_time)

        with self.assertRaises(exceptions.Forbidden):
            other.delete()

        other.event_based_hold = False
        other.patch()
        other.delete()

        bucket.default_event_based_hold = False
        bucket.patch()

        self.assertFalse(bucket.default_event_based_hold)
        self.assertIsNone(bucket.retention_period)
        self.assertIsNone(bucket.retention_policy_effective_time)
        self.assertFalse(bucket.retention_policy_locked)

        blob.upload_from_string(payload)
>       self.assertFalse(blob.event_based_hold)
E       AssertionError: True is not false

tests/system/test_system.py:2442: AssertionError
```

--- a/tests/system/test_system.py
+++ b/tests/system/test_system.py
@@ -42,6 +42,7 @@ import google.api_core
 from google.api_core import path_template
 import google.oauth2
 from test_utils.retry import RetryErrors
+from test_utils.retry import RetryInstanceState
 from test_utils.system import unique_resource_id
 from test_utils.vpcsc_config import vpcsc_config
 
@@ -56,6 +57,9 @@ def _bad_copy(bad_request):
     err_msg = bad_request.message
     return err_msg.startswith("No file found in request. (POST") and "copyTo" in err_msg
 
+def _no_event_based_hold(blob):
+    return not blob.event_based_hold
+
 
 retry_429 = RetryErrors(exceptions.TooManyRequests, max_tries=6)
 retry_429_harder = RetryErrors(exceptions.TooManyRequests, max_tries=10)
@@ -63,6 +67,7 @@ retry_429_503 = RetryErrors(
     [exceptions.TooManyRequests, exceptions.ServiceUnavailable], max_tries=10
 )
 retry_bad_copy = RetryErrors(exceptions.BadRequest, error_predicate=_bad_copy)
+retry_no_event_based_hold = RetryInstanceState(_no_event_based_hold)
 
 
 def _empty_bucket(client, bucket):
@@ -2482,6 +2487,11 @@ class TestRetentionPolicy(unittest.TestCase):
         self.assertFalse(bucket.retention_policy_locked)
 
         blob.upload_from_string(payload)
+
+        # https://github.com/googleapis/python-storage/issues/435
+        if blob.event_based_hold:
+            retry_no_event_based_hold(blob.reload)()
+
         self.assertFalse(blob.event_based_hold)
         self.assertFalse(blob.temporary_hold)
         self.assertIsNone(blob.retention_expiration_time)

--- a/tests/system/test_system.py
+++ b/tests/system/test_system.py
@@ -57,6 +57,7 @@ def _bad_copy(bad_request):
     err_msg = bad_request.message
     return err_msg.startswith("No file found in request. (POST") and "copyTo" in err_msg
 
+
 def _no_event_based_hold(blob):
     return not blob.event_based_hold
 

